rust/osm2lanes/src/transform/tags_to_lanes/lane_count.rs

```diff
@@ -0,0 +1,375 @@
+                },
+                (Some(l), None, Some(b)) => Self {
+                    lanes: Infer::Direct(l),
+                    forward: Infer::Calculated(
+                        l.checked_sub(b)
+                            .unwrap()
+                            .checked_sub(both_way_lanes)
+                            .unwrap(),
+                    ),
+                    backward: Infer::Direct(b),
+                    both_ways,
+                },
+                // Alleyways or narrow unmarked roads, usually:
+                (Some(1), None, None) => Self {
+                    lanes: Infer::Direct(1),
+                    forward: Infer::Default(0),
+                    backward: Infer::Default(0),
+                    both_ways: Infer::Default(1),
+                },
+                (Some(l), None, None) => {
+                    if l % 2 == 0 && centre_turn_lane.0.unwrap_or(false) {
+                        // Only tagged with lanes and deprecated center_turn_lane tag.
+                        // Assume the center_turn_lane is in addition to evenly divided lanes.
+                        Self {
+                            lanes: Infer::Calculated(l.checked_add(1).unwrap()),
+                            forward: Infer::Default(l.checked_div(2).unwrap()),
+                            backward: Infer::Default(l.checked_div(2).unwrap()),
+                            both_ways: Infer::Calculated(1),
+                        }
+                    } else {
+                        // Distribute normal lanes evenly.
+                        let remaining_lanes = l.checked_sub(both_way_lanes).unwrap();
+                        if remaining_lanes % 2 != 0 {
+                            warnings.push(TagsToLanesMsg::ambiguous("Total lane count cannot be evenly divided between the forward and backward",
+                                tags.subset(&[
+                                    "lanes",
+                                    "lanes:both_ways",
+                                ]),
+                            ));
+                        }
+                        let half = remaining_lanes
+                            .checked_add(1)
+                            .unwrap()
+                            .checked_div(2)
+                            .unwrap(); // usize division rounded up.
+                        Self {
+                            lanes: Infer::Direct(l),
+                            forward: Infer::Direct(half),
+                            backward: Infer::Direct(remaining_lanes.checked_sub(half).unwrap()),
+                            both_ways,
+                        }
+                    }
+                },
+                (None, _, _) => {
+                    // Without the "lanes" tag, assume one normal lane in each dir
+                    // Adding any bus:lanes as needed.
+                    const DEFAULT: usize = 1;
+                    let tagged_bus_lanes: Option<usize> =
```

Comment:
I'm a bit lost between this and #111 -- I thought we had a more consolidated struct just to handle bus stuff, and we would reconcile / merge the results of that here?

Reply:
I go a different route, for better or worse :/